src/codegen/compile.ml

```diff
@@ -272,20 +273,17 @@ module Const = struct
   type v =
     | Fun of (unit -> int32) * fun_rhs (* function pointer calculated upon first use *)
     | Message of int32 (* anonymous message, only temporary *)
-    | Obj of (string * t) list
+    | Obj of (string * v) list
     | Unit
-    | Array of t list (* also tuples, but not nullary *)
-    | Tag of (string * t)
-    | Opt of t
+    | Array of v list (* also tuples, but not nullary *)
+    | Tag of (string * v)
+    | Opt of v
     | Lit of lit
 
-  (* A constant known value together with a vanilla pointer.
-     Typically a static memory location, could be an unboxed scalar.
-     Filled on demand.
-   *)
-  and t = (int32 Lib.Promise.t * v)
-
-  let t_of_v v = (Lib.Promise.make (), v)
+  let eq v1 v2 = match v1, v2 with
+    | Lit l1, Lit l2 -> lit_eq l1 l2
+    | Fun (f1_fp, _), Fun (f2_fp, _) -> f1_fp() = f2_fp()
```

Comment:
I wonder if this equality test is a good idea. I think it might wind up generating code that is no necessarily needed.
Maybe it would be safe to add a gen-syme'd id field, and compare that (or use f1_fp == f2_fp for physical equality)

Reply:
Thanks for noticing. I corrected it by using the `id` field approach.